app/graphql/resolvers/waste_location_search.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require "search_object/plugin/graphql"
+require "graphql/query_resolver"
+
+class Resolvers::WasteLocationSearch
+  include SearchObject.module(:graphql)
+
+  scope { Address.where(id: Waste::LocationType.all.pluck(:address_id)) }
+
+  type types[Types::AddressType]
+
+  class WasteLocationOrder < ::Types::BaseEnum
+    value "createdAt_ASC"
+    value "createdAt_DESC"
+    value "updatedAt_ASC"
+    value "updatedAt_DESC"
+    value "id_ASC"
+    value "id_DESC"
+  end
+
+  option :limit, type: types.Int, with: :apply_limit
+  option :skip, type: types.Int, with: :apply_skip
+  option :ids, type: types[types.ID], with: :apply_ids
+  option :order, type: WasteLocationOrder, default: "createdAt_DESC"
+
+  def apply_limit(scope, value)
+    scope.limit(value)
+  end
+
+  def apply_skip(scope, value)
+    scope.offset(value)
+  end
+
+  def apply_ids(scope, value)
+    scope.where(id: value)
+  end
+
+  def apply_order(scope, value)
+    scope.order(value)
+  end
+
+  def apply_order_with_created_at_desc(scope)
+    scope.order("created_at DESC")
+  end
+
+  def apply_order_with_created_at_asc(scope)
+    scope.order("created_at ASC")
+  end
+
+  def apply_order_with_updated_at_desc(scope)
+    scope.order("updated_at DESC")
+  end
+
+  def apply_order_with_updated_at_asc(scope)
+    scope.order("updated_at ASC")
+  end
+
+  def apply_order_with_id_desc(scope)
+    scope.order("id DESC")
+  end
+
+  def apply_order_with_id_asc(scope)
+    scope.order("id ASC")
+  end
```

Comment:
wir sollten mal überlegen die standard order methoden in eine klasse zu packen von der alle erben, sodass wir das nicht immer copy&pasten in jeden neuen resolver?!

Reply:
ja, das wäre in der Tat gut!

Reply:
issue dazu ist erstellt